src/java/org/apache/cassandra/cql3/restrictions/RestrictionSet.java

```diff
@@ -418,21 +431,23 @@ private void addRestrictionForColumns(List<ColumnMetadata> columnDefs, SingleRes
                     lastRestrictionColumn = column;
                     lastRestriction = restriction;
                 }
+                if (clear)
+                    newRestrictions.removeAll(column);
```

Comment:
Why is this necessary/

Reply:
The `newRestrictions` has changed from a `HashMap` to an `ArrayListMultiMap` because with disjunction we can have more than one restriction per column.

As a result, when we are replacing merged restrictions for non-disjunct queries we need to clear the entry before putting the new merged restriction into the `MultiMap`.

I have renamed the `clear` to `replace` to make this clearer and I have added some comments around this.

Reply:
+1